### PR TITLE
update eligibility service to use direction methods

### LINKED ISSUE #316: Recognise movements indicated as 'A' or 'D'
The official dataset used the strings 'A' and 'D' to indicate arrivals and departures in the movements table. Our existing code expects 'arrival' or 'departure', so it wasn't detecting the new records.

Our code should be updated to recognise both 'arrival' and 'A' for arrivals, and 'departure' and 'D' for departures.

See eligibility_service.rb:84 and ALSO eligibility_service.rb:98 where the same check is used for visa status.

Acceptance criteria
- [x] There is a new test scenario where the movement data uses 'A' and 'D', and the client is recognised as eligible for citizenship

Bonus points:
- [x] Update the Movement model to have methods `arrival?` and `departure?` which return true if the movement is an arrival or a departure respectively, and use that instead of checking strings in the service code.
- [x] Add a validation to the Movement model to ensure that the `direction` is one of the expected values ('arrival', 'A', 'departure', 'D')
- [x] Update the Movements factory to use 'A' and 'D' so it's more like the real data.

### PR DESCRIPTION
**Pull request by dlouise64**
Using this branch since other one closed and couldn't reopen it!

closes #316

### CODE REVIEW

File: app/services/eligibility_service.rb [hunk 1]
```diff
@@ -81,9 +81,9 @@ def presence_values
     presence = {}
 
     @client.movements.order(:carrier_date_time).each do |movement|
-      if movement.direction == 'arrival'
+      if movement.arrival?(movement.direction)
```
Comment (Br3nda) [inline app/services/eligibility_service.rb]: why do we need to pass this is? Can't the `.arrival?` method read that member variable itself?

Comment (Br3nda) [reply app/services/eligibility_service.rb]: how about just doing:
`if movement.arrival?`

File: spec/models/movement_spec.rb [hunk 2]
```diff
@@ -3,9 +3,16 @@
...
 
   it 'has a factory which produces valid models by default' do
     expect(subject).to be_valid
   end
-end
+
+  it 'recognises when movement direction is A or D' do
+    expect(arrival).to be_valid
+    expect(departure).to be_valid
+  end
```
Comment (Br3nda) [inline spec/models/movement_spec.rb]: how about a test for this
`expect(arrival.arrival?) to be true)`

File: app/services/eligibility_service.rb (hunk 1 above)
Comment (dlouise64) [reply app/services/eligibility_service.rb]: @Br3nda Oops! yep got it :)

File: spec/models/movement_spec.rb [hunk 3]
```diff
@@ -3,9 +3,17 @@
...
 
   it 'has a factory which produces valid models by default' do
     expect(subject).to be_valid
   end
-end
+
+  it 'recognises when movement direction is A or D' do
+    expect(arrival).to be_valid
+    expect(departure).to be_valid
+    expect(arrival.arrival?).to be true
```
Comment (Br3nda) [inline spec/models/movement_spec.rb]: What about the rest of the methods?
```
expect(departure.departure?) to eq true
expect(arrival.departure?) to eq false
expect(departure.arrival?) to eq false
```

**Pull request merged**